src/expression/ekat_expression_compare.hpp

```diff
@@ -0,0 +1,149 @@
+#ifndef EKAT_EXPRESSION_COMPARE_HPP
+#define EKAT_EXPRESSION_COMPARE_HPP
+
+#include "ekat_expression_base.hpp"
+
+#include "ekat_std_utils.hpp"
+#include "ekat_kernel_assert.hpp"
+#include "ekat_assert.hpp"
+
+namespace ekat {
+
+enum class Comparison : int {
+  EQ,   // ==
+  NE,   // !=
+  GT,   // >
+  GE,   // >=
+  LT,   // <
+  LE    // <=
+};
+
+template<typename ELeft, typename ERight>
+class CmpExpression : public Expression<CmpExpression<ELeft,ERight>> {
+public:
+  using ret_t = int;
```

Comment:
I used `int` for some reason, but it could probably be changed into bool. I already have a follow up branch where I extend the expressions framework to work with Packs, so it can be changed there.